core/cloudflow-runner/src/main/scala/cloudflow/runner/Runner.scala

```diff
@@ -68,32 +68,41 @@ object Runner extends RunnerConfigResolver with StreamletLoader {
         val streamletExecution = loadedStreamlet.streamlet.run(withPodRuntimeConfig)
         loadedStreamlet.streamlet.logStartRunnerMessage(formatBuildInfo)
 
-        Try {
-          // the runner waits for the execution to complete
-          // In normal circumstances it will run forever for streaming data source unless
-          // being stopped forcibly or any of the queries faces an exception
-          Await.result(streamletExecution.completed, Duration.Inf)
-        } match {
-          case Success(_) ⇒ System.exit(0)
-          case Failure(ex @ ExceptionAcc(exceptions)) ⇒
+        // the runner waits for the execution to complete
+        // In normal circumstances it will run forever for streaming data source unless
+        // being stopped forcibly or any of the queries faces an exception
+        Await.ready(streamletExecution.completed, Duration.Inf)
+        streamletExecution.completed.value match {
+          case None => // can't happen b/c we wait for it to be ready
+            log.error("Unexpected streamletExecution state: No result after termination. This is most probably a bug.")
+            shutdown(loadedStreamlet, Some(new IllegalStateException("Unexpected streamletExecution state: No result after termination")))
+          case Some(Success(_)) ⇒ shutdown(loadedStreamlet)
+          case Some(Failure(ex @ ExceptionAcc(exceptions))) ⇒
             exceptions.foreach(ErrorEvents.report(loadedStreamlet, withPodRuntimeConfig, _))
-            shutdownWithFailure(loadedStreamlet, ex)
-          case Failure(ex) ⇒
+            shutdown(loadedStreamlet, Some(ex))
+          case Some(Failure(ex)) ⇒
             ErrorEvents.report(loadedStreamlet, withPodRuntimeConfig, ex)
-            shutdownWithFailure(loadedStreamlet, ex)
+            shutdown(loadedStreamlet, Some(ex))
         }
       case Failure(ex) ⇒ throw new Exception(ex)
     }
   }
 
-  private def shutdownWithFailure(loadedStreamlet: LoadedStreamlet, ex: Throwable) = {
+  private def shutdown(loadedStreamlet: LoadedStreamlet, maybeException: Option[Throwable] = None) = {
     // we created this file when the pod started running (see AkkaStreamlet#run)
     Files.deleteIfExists(
       Paths.get(s"/tmp/${loadedStreamlet.config.streamletRef}.txt")
     )
-    log.error("Fatal error has occurred:", ex)
+    maybeException
+      .map { ex =>
+        log.error("Fatal error has occurred:", ex)
+        System.exit(-1)
+      }
+      .getOrElse {
+        log.info("Streamlet terminating due to successful termination of query:")
```

Comment:
This message is specific to spark (query), should not be here? Also ends on colon?